### DISCUSSION TITLE
Scaling several deployments with single HTTPScaledObject

### DISCUSSION THREAD

**Discussion by dimber-cais** [kedacore/http-add-on#405] [Q&A]
I have a use case whereby as part of a CI process, we create a new K8 namespace and add several deployments within it. One of the deployments is the UI facing application and the others are backend micro-services. 

We'd like these temporary environments to scale to zero when not in use. So ideally we would scale all the deployments to zero when there's no requests to the UI facing application, and they all scale up when there are requests.

Trying to work out if that's possible with Keda and Keda HTTP?

**Comment by dimber-cais** [comment]
Doing a bit more digging, perhaps I could use `HttpScaledObject` for the UI application deployment, and then use the `kubernetes-workload` scaler for the other deployments, so that they are scaled based on the UI application. 

Is that a reasonable approach here?

**Reply by arschles** [reply]
@dimber-cais the way that you described would work, but assuming the other backend-microservices accept HTTP traffic and the frontend calls them via a stable name -- like internal service DNS -- from the UI frontend, you could consider:

- creating an `HTTPScaledObject` for each of them
- setting the `host` to the name of the `Service`

This way may be a bit more straightforward, since you'll be scaling each backend service based on incoming traffic to just that service, rather than scaling all of the backends after experiencing incoming traffic on the frontend. Let me know what you think.

**Reply by surachai-tep** [reply]
@arschles I have similar deployment like @dimber-cais. We have deployed more than 1 deployments in a single namespace and use url base routing like mydomain.com/app1, mydomain.com/app2, etc. 
How can I setup http-adds-on on this scenario?

Thanks

**Reply by arschles** [reply]
@surachai-tep I think you can use the same scheme as what I wrote in https://github.com/kedacore/http-add-on/discussions/405#discussioncomment-2240147. Does that look like it will work for you?

**Reply by surachai-tep** [reply]
Hi @arschles , I don't really understand the "setting the host to the name of the Service" step. Could you please explain more how the host looks like in each HTTPScaledObject ?

Thank you

**Reply by arschles** [reply]
@surachai-tep it's actually independent of the HTTP Addon altogether. For example, say you have a frontend talking to a backend that is running behind a standard Kubernetes `Service`, and that `Service` is called `myBackendSvc`.

Assuming you have standard Kubernetes in-cluster DNS properly configured, your frontend can make HTTP requests to the interceptor, with `myBackendSvc` in the `Host` header.  That means you can create an `HTTPScaledObject` for your backend with `myBackendSvc` as the `host` value. would that work for you? if not, let me know why not and we can figure out another way, or add a new feature to support this for you.
